### PR TITLE
fix: Icon in disabled Button shouldn't change color on hover or pressed

### LINKED ISSUE #29341: [Bug]: Icon in disabled subtle Button changes color on hover
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz
    Memory: 9.20 GB / 31.60 GB
  Browsers:
    Edge: Spartan (44.22621.2283.0), Chromium (116.0.1938.81)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://react.fluentui.dev/?path=/docs/components-button-button--default

### Bug Description

Hover Button component with icon and `appearance="subtle"` and disabled state

## Actual Behavior
image

## Expected Behavior
image

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by ValentinaKozlova**
Hover Button component with icon and appearance="subtle" and disabled state

## Previous Behavior
image

## New Behavior
image

- Fixes #29341

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/InfoButton-mount_base.html">9</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/InfoButton-mount.html">17</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/Avatar-mount_base.html">625</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/Avatar-mount.html">620</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/Button-mount_base.html">328</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/Button-mount.html">313</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/Field-mount_base.html">1114</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/Field-mount.html">1113</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/FluentProvider-mount_base.html">690</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/FluentProvider-mount.html">695</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/FluentProviderWithTheme-mount_base.html">81</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/FluentProviderWithTheme-mount.html">87</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">60</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">65</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">85</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">81</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/InfoButton-mount_base.html">9</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/InfoButton-mount.html">17</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/MakeStyles-mount_base.html">874</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/MakeStyles-mount.html">857</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/Persona-mount_base.html">1723</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/Persona-mount.html">1694</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/SpinButton-mount_base.html">1368</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29342/perf-test-react-components/SpinButton-mount.html">1353</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                                             |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                  Change |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                                       |  `83.657 kB`<br />`22.915 kB` |  `83.765 kB`<br />`22.943 kB` | `108 B` <br />`28 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                                                                    |  `39.444 kB`<br />`10.738 kB` |  `39.552 kB`<br />`10.766 kB` | `108 B` <br />`28 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                                                                    |  `46.799 kB`<br />`12.233 kB` |  `46.907 kB`<br />`12.261 kB` | `108 B` <br />`28 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                                                            |   `44.218 kB`<br />`12.12 kB` |  `44.326 kB`<br />`12.148 kB` | `108 B` <br />`28 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                                                          |  `52.232 kB`<br />`13.705 kB` |   `52.34 kB`<br />`13.731 kB` | `108 B` <br />`26 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                                                                        |   `56.49 kB`<br />`12.641 kB` |  `56.598 kB`<br />`12.667 kB` | `108 B` <br />`26 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `69.227 kB`<br />`19.584 kB` |  `69.335 kB`<br />`19.614 kB` | `108 B` <br />`30 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `206.783 kB`<br />`59.085 kB` | `206.891 kB`<br />`59.113 kB` | `108 B` <br />`28 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                  |         Size (minified/GZIP) |
| -------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                                     | `91.484 kB`<br />`26.485 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                               |  `86.31 kB`<br />`24.946 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                                   |  `12.074 kB`<br />`5.034 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                                   |    `14.323 kB`<br />`5.8 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                                 |  `13.031 kB`<br />`5.404 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `40.793 kB`<br />`13.521 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                              |    `6.48 kB`<br />`2.203 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/b5f4971b4b04d5a02983ec472c4cc565ebd98d80'>b5f4971b4b04d5a02983ec472c4cc565ebd98d80</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-button/src/components/Button/useButtonStyles.styles.ts [hunk 1]
```diff
@@ -332,6 +332,9 @@ const useRootDisabledStyles = makeStyles({
       [`& .${iconRegularClassName}`]: {
         display: 'inline',
       },
+      [`&:disabled .${buttonClassNames.icon}`]: {
+        color: tokens.colorNeutralForegroundDisabled,
+      },
```
Comment (khmakoto) [inline packages/react-components/react-button/src/components/Button/useButtonStyles.styles.ts:337]: For completeness you should probably also add this same style in the base (around line 321 below the `cursor: 'not-allowed'` line) and `:hover:active` states (around line 352 similar to where you're putting this one but for `:hover:active`).

Another thing is that you don't really need the `disabled` style here (I doubled checked locally).

```suggestion
      [`& .${buttonClassNames.icon}`]: {
        color: tokens.colorNeutralForegroundDisabled,
      },
```

File: change/@fluentui-react-button-d89ce90d-cbf8-4b16-9a02-2d6606cea56e.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "fix: Icon in disabled Button shouldn't change color on hover",
```
Comment (khmakoto) [inline change/@fluentui-react-button-d89ce90d-cbf8-4b16-9a02-2d6606cea56e.json:3]: _nit:_ 

```suggestion
  "comment": "fix: Icon in disabled Button shouldn't change color on hover or pressed.",
```

**Review by khmakoto — APPROVED** [review]

**Pull request merged**